src/java.base/share/classes/java/lang/foreign/SymbolLookup.java

```diff
@@ -144,9 +144,39 @@ public interface SymbolLookup {
      * @param name the symbol name
      * @return a zero-length memory segment whose address indicates the address of
      *         the symbol, if found
+     * @see #findOrThrow(String)
      */
     Optional<MemorySegment> find(String name);
 
+    /**
+     * {@return the address of the symbol with the provided {@code name} or throws an
+     *          {@linkplain IllegalArgumentException} if no such address can be found}
+     *<p>
+     * This is a convenience method that provides better exception messages compared
+     * to:
+     * {@snippet lang= java :
+     *    MemorySegment address = lookup.find("foo")
+     *        .orElseThrow(IllegalArgumentException::new);
+     * }
+     *
+     * @param name the symbol name to look up
```

Comment:
'symbol' and `name` or almost synonymous. A symbol identifies a function or global variable in a library. So maybe this could say
```suggestion
     * @param symbol the symbol to look up
```

(and then change the name of the parameter).